pkg/lwip/patches/0002-adapt-netdb-functions-for-RIOT.patch

```diff
@@ -0,0 +1,249 @@
+From 51eb852671389157892781db7f80c39894d39996 Mon Sep 17 00:00:00 2001
+From: Hendrik van Essen <hendrik.ve@fu-berlin.de>
+Date: Tue, 15 Jun 2021 22:05:13 +0200
+Subject: [PATCH] adapt netdb functions for RIOT
+
+---
+ src/api/netdb.c                  | 23 ++++++++++-------------
+ src/include/lwip/inet.h          | 28 ++++++++++++++++++++++++++++
+ src/include/lwip/netdb.h         | 19 ++++---------------
+ src/include/lwip/priv/memp_std.h |  4 ++--
+ 4 files changed, 44 insertions(+), 30 deletions(-)
+
+diff --git a/src/api/netdb.c b/src/api/netdb.c
+index 87714259..4abd26e8 100644
+--- a/src/api/netdb.c
++++ b/src/api/netdb.c
+@@ -37,7 +37,7 @@
+ 
+ #include "lwip/netdb.h"
+ 
+-#if LWIP_DNS && LWIP_SOCKET
++#if LWIP_NETDB
+ 
+ #include "lwip/err.h"
+ #include "lwip/mem.h"
+@@ -49,6 +49,8 @@
+ #include <string.h> /* memset */
+ #include <stdlib.h> /* atoi */
+ 
++#include "net/sock/dns.h"
++
+ /** helper struct for gethostbyname_r to access the char* buffer */
+ struct gethostbyname_r_helper {
+   ip_addr_t *addr_list[2];
+@@ -86,7 +88,6 @@ int h_errno;
+ struct hostent *
+ lwip_gethostbyname(const char *name)
+ {
+-  err_t err;
+   ip_addr_t addr;
+ 
+   /* buffer variables for lwip_gethostbyname() */
+@@ -97,8 +98,8 @@ lwip_gethostbyname(const char *name)
+   HOSTENT_STORAGE char s_hostname[DNS_MAX_NAME_LENGTH + 1];
+ 
+   /* query host IP address */
+-  err = netconn_gethostbyname(name, &addr);
+-  if (err != ERR_OK) {
++  int res = sock_dns_query(name, &addr, AF_INET);
++  if (res < 0) {
```

Comment:
I don't quite understand this: why can't `netconn_gethostbyname()` be used here?